File: 백준/Silver/25192. 인사성 밝은 곰곰이/인사성 밝은 곰곰이.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let lines = [];
rl.on('line',(line)=>{
    lines.push(line.trim());
});

rl.on('close',()=>{
    const N = Number(lines[0]);
    lines.shift();
    let cnt = 0;
    let set = new Set();
    for (let i = 1; i < N; i++){
        const temp = lines[i];
        if (temp === "ENTER") {
            cnt += set.size;
            set.clear();
        }else{
            set.add(temp);
        }
        if(i === N-1) {
            cnt += set.size;
        }
    }
    console.log(cnt);
});

File: 백준/Silver/25192. 인사성 밝은 곰곰이/인사성 밝은 곰곰이.py
import sys

N = int(sys.stdin.readline().strip())
sys.stdin.readline().strip()
cnt = 0
char_set = set()
for i in range(1,N):
    temp = sys.stdin.readline().strip()
    if temp == "ENTER":
        cnt += len(char_set)
        char_set.clear()
    else:
        char_set.add(temp)
    if i == N-1:
        cnt += len(char_set)
print(cnt)